pubtools/_quay/push_docker.py

```diff
@@ -139,7 +142,13 @@ def get_docker_push_items(self):
             if not item.metadata.get("pull_data"):
                 raise BadPushItem("Push item {0} doesn't contain pull data.".format(item))
             LOG.info("Docker push item found: {0}".format(item))
-            docker_push_items.append(item)
+            pull_url = item.metadata["pull_url"]
+            if pull_url in url_items:
+                url_items[pull_url].append(item)
+            else:
+                url_items[pull_url] = [item]
+        for _, items in sorted(url_items.items()):
+            docker_push_items.append(items[0])
```

Comment:
Can't push items have other differences than pull url? For example ["tags"] field tells us where to push the image. We would this information if we filtered items only by unique pull url.

Or am I misunderstanding something?

Reply:
They have difference on some other metadata, e.g. dev_tags, arch, but not on tags.

Reply:
@midnightercz you probably understand best how sources generate push items, is this safe?

Reply:
Container push items are populated here: 
https://github.com/release-engineering/pushsource/blob/master/src/pushsource/_impl/backend/koji_source.py#L403
Which is also where this filter should be implemented. Functionality is split into source/target so having this implemented here mixes those to together.

Reply:
But besides that, change is ok. Destination tags are set in target, so if push items share source url, it's ok to filter duplicates

Reply:
I doubt it's wanted filter for Pulp targets. Pushes to pulp pull images from url (digest of each arch) in image archive , not the url (manifest list) in build information.